packages/runner/src/suite.ts

```diff
@@ -77,7 +77,7 @@ function createSuiteCollector(name: string, factory: SuiteFactory = () => { }, m
       mode,
       suite: undefined!,
       fails: this.fails,
-      retry: options?.retry,
+      retry: options?.retry ?? runner.config.retry,
```

Comment:
```suggestion
      retry: options?.retry || runner.config.retry,
```

I don't think it makes sense to be 0 here, so feel free to use `||`

Reply:
Why doesn't it makes sense to you? 0 is the default. If someone wants to override it, it should be possible to override

Reply:
If the retry is 0, the test will not run. this may lead to some misunderstandings. In addition, the default should be 1.

Reply:
Why wouldn't it run? It should retry only if actual run failed. This is how it works in other frameworks. (`retry: 1` calls test _twice_ - one actual test, the second one is the retry)

Reply:
You can see here that the retry is ensured at least once, if it is 0, the for loop will not be executed and the test will be skipped.
https://github.com/vitest-dev/vitest/blob/0c498dab10cf6a40742fb1136bf6e2b3f51635f6/packages/runner/src/run.ts#L138-L140

Also, now that retry has a default value, we can delete line 38

Reply:
Then it is a bug in Vitest implementation, retry should be 0 if test didn't fail

Reply:
Yes, I aggre with you, this looks a bit confusing to me at the beginning

Reply:
We should fix it first then, before adding a global option

Reply:
Can i do it？